### PR TITLE
add kbdpadmk3

### PR DESCRIPTION
**Pull request by moyi4681**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/kbdpad/mk3/mk3.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+4,4,4,
+4,4,4,4,
+4,4
+} };
+
+
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void)
+{
+    if (IS_HOST_LED_ON(USB_LED_NUM_LOCK))
```
Comment (fauxpark) [inline keyboards/kbdfans/kbdpad/mk3/mk3.c:46]:
```suggestion
    if (host_keyboard_led_state().num_lock)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kbdfans/kbdpad/mk3/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    _______, _______
+  ),
+  [3] = LAYOUT_numpad_6x4(
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+    _______, _______, _______, _______,
+    _______, _______
+  ),
```
Comment (zvecr) [inline keyboards/kbdfans/kbdpad/mk3/keymaps/default/keymap.c:51]:
```suggestion
  [0] = LAYOUT_numpad_6x4( /* Base */
    KC_ESC,  KC_LCTL, KC_LALT, KC_BSPC,
    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
    KC_P7,   KC_P8,   KC_P9,
    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
    KC_P1,   KC_P2,   KC_P3,
    KC_P0,            KC_PDOT, KC_PENT,
  )
```

File: keyboards/kbdfans/kbdpad/mk3/mk3.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_numpad_6x4( \
+    K00, K01, K02, K03, \
+    K10, K11, K12, K13, \
+    K20, K21, K22, K23, \
+    K30, K31, K32, \
+    K40, K41, K42, K43, \
+    K50,      K52 \
```
Comment (zvecr) [inline keyboards/kbdfans/kbdpad/mk3/mk3.h:34]:
```suggestion
    K00, K01, K02, K03, \
    K10, K11, K12, K13, \
    K20, K21, K22, \
    K30, K31, K32, K23, \
    K40, K41, K42, \
    K50,      K52, K43, \
```

File: keyboards/kbdfans/kbdpad/mk3/info.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                {"x":3, "y":2, "h":2},
+
+                {"x":0, "y":4},
+                {"x":1, "y":4},
+                {"x":2, "y":4},
+                
+                {"x":0, "y":5, "w":2},
+                {"x":2, "y":5},
+                {"x":3, "y":4, "h":2}
+            ]
```
Comment (zvecr) [inline keyboards/kbdfans/kbdpad/mk3/info.json:51]:
```suggestion
            "layout": [
                {"x":0, "y":0},
                {"x":1, "y":0},
                {"x":2, "y":0},
                {"x":3, "y":0},

                {"x":0, "y":1},
                {"x":1, "y":1},
                {"x":2, "y":1},
                {"x":3, "y":1},

                {"x":0, "y":2},
                {"x":1, "y":2},
                {"x":2, "y":2},

                {"x":0, "y":3},
                {"x":1, "y":3},
                {"x":2, "y":3},
                {"x":3, "y":2, "h":2},

                {"x":0, "y":4},
                {"x":1, "y":4},
                {"x":2, "y":4},

                {"x":0, "y":5, "w":2},
                {"x":2, "y":5},
                {"x":3, "y":4, "h":2}
            ]
```

File: keyboards/kbdfans/kbdpad/mk3/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    _______, _______
+  ),
+  [3] = LAYOUT_numpad_6x4(
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, _______,
+    _______, _______, _______, 
+    _______, _______, _______, _______,
+    _______, _______
+  ),
```
Comment (zvecr) [inline keyboards/kbdfans/kbdpad/mk3/keymaps/via/keymap.c:51]:
```suggestion
  [0] = LAYOUT_numpad_6x4( /* Base */
    KC_ESC,  KC_LCTL, KC_LALT, KC_BSPC,
    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
    KC_P7,   KC_P8,   KC_P9,
    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
    KC_P1,   KC_P2,   KC_P3,
    KC_P0,            KC_PDOT, KC_PENT,
  ),
  [1] = LAYOUT_numpad_6x4(
    _______, _______, _______, _______,
    _______, _______, _______, _______,
    _______, _______, _______,
    _______, _______, _______, _______, 
    _______, _______, _______,
    _______,          _______, _______
  ),
  [2] = LAYOUT_numpad_6x4(
    _______, _______, _______, _______,
    _______, _______, _______, _______,
    _______, _______, _______,
    _______, _______, _______, _______, 
    _______, _______, _______,
    _______,          _______, _______
  ),
  [3] = LAYOUT_numpad_6x4(
    _______, _______, _______, _______,
    _______, _______, _______, _______,
    _______, _______, _______,
    _______, _______, _______, _______, 
    _______, _______, _______,
    _______,          _______, _______
  ),
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbdpad/mk3/mk3.h [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_numpad_6x4( \
+    K00, K01, K02, K03, \
+    K10, K11, K12, K13, \
+    K20, K21, K22, \
+    K30, K31, K32, K23, \
+    K40, K41, K42, \
+    K50,      K52, K43, \
```
Comment (fauxpark) [inline keyboards/kbdfans/kbdpad/mk3/mk3.h:34]:
```suggestion
    K50,      K52, K43 \
```

File: keyboards/kbdfans/kbdpad/mk3/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_numpad_6x4( /* Base */
+    KC_ESC,  KC_LCTL, KC_LALT, KC_BSPC,
+    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+    KC_P7,   KC_P8,   KC_P9,
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+    KC_P1,   KC_P2,   KC_P3,
+    KC_P0,            KC_PDOT, KC_PENT,
```
Comment (fauxpark) [inline keyboards/kbdfans/kbdpad/mk3/keymaps/via/keymap.c:26]:
```suggestion
    KC_P0,            KC_PDOT, KC_PENT
```

File: keyboards/kbdfans/kbdpad/mk3/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_numpad_6x4( /* Base */
+    KC_ESC,  KC_LCTL, KC_LALT, KC_BSPC,
+    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+    KC_P7,   KC_P8,   KC_P9,
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+    KC_P1,   KC_P2,   KC_P3,
+    KC_P0,            KC_PDOT, KC_PENT,
```
Comment (fauxpark) [inline keyboards/kbdfans/kbdpad/mk3/keymaps/default/keymap.c:26]:
```suggestion
    KC_P0,            KC_PDOT, KC_PENT
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbdpad/mk3/mk3.c [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+4,4,4,4,
+4,4,4,4,
+4,4,4,
+4,4,4,4,
+4,4
+} };
+
+
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void)
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad/mk3/mk3.c:44]: Note to self: #18450

**Review by drashna — DISMISSED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kbdfans/kbdpad/mk3/mk3.c (hunk 9 above)
Comment (drashna) [reply keyboards/kbdfans/kbdpad/mk3/mk3.c:44]:
```suggestion
bool rgb_matrix_indicators_kb(void) {
    if (!rgb_matrix_indicators_user()) { 
        return false;
    }
```

File: keyboards/kbdfans/kbdpad/mk3/mk3.c [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+{
+    if (host_keyboard_led_state().num_lock)
+    {
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad/mk3/mk3.c:47]:
```suggestion
    if (host_keyboard_led_state().num_lock) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbdpad/mk3/mk3.c [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+__attribute__ ((weak))
+bool rgb_matrix_indicators_kb(void) {
+    if (!rgb_matrix_indicators_user()) { 
+        return false;
+    }
+    if (host_keyboard_led_state().num_lock) {
+        rgb_matrix_set_color(7, 0xFF, 0xFF, 0xFF);
+    }
+}
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad/mk3/mk3.c:51]:
```suggestion
    return true;
}
```

File: keyboards/kbdfans/kbdpad/mk3/mk3.c [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+4,4,4,4,
+4,4,4,4,
+4,4,4,4,
+4,4,4,
+4,4,4,4,
+4,4
+} };
+
+
+__attribute__ ((weak))
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad/mk3/mk3.c:43]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/kbdfans/kbdpad/mk3/config.h [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+#define RGB_MATRIX_LED_COUNT 21
```
Comment (drashna) [inline keyboards/kbdfans/kbdpad/mk3/config.h:17]:
```suggestion
#pragma once

#define RGB_MATRIX_LED_COUNT 21
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**